include/seqan3/core/simd/concept.hpp

```diff
@@ -48,12 +48,12 @@ concept simd_concept = requires (simd_t a, simd_t b)
     { a[0] } -> typename simd_traits<simd_t>::scalar_type;
 
     // require comparison operators
-    requires std::Same<decltype(a == b), typename simd_traits<simd_t>::mask_type>;
-    requires std::Same<decltype(a != b), typename simd_traits<simd_t>::mask_type>;
-    requires std::Same<decltype(a <  b), typename simd_traits<simd_t>::mask_type>;
-    requires std::Same<decltype(a >  b), typename simd_traits<simd_t>::mask_type>;
-    requires std::Same<decltype(a <= b), typename simd_traits<simd_t>::mask_type>;
-    requires std::Same<decltype(a >= b), typename simd_traits<simd_t>::mask_type>;
+    requires sizeof(decltype(a == b)) == sizeof(typename simd_traits<simd_t>::mask_type);
+    requires sizeof(decltype(a != b)) == sizeof(typename simd_traits<simd_t>::mask_type);
+    requires sizeof(decltype(a <  b)) == sizeof(typename simd_traits<simd_t>::mask_type);
+    requires sizeof(decltype(a >  b)) == sizeof(typename simd_traits<simd_t>::mask_type);
+    requires sizeof(decltype(a <= b)) == sizeof(typename simd_traits<simd_t>::mask_type);
+    requires sizeof(decltype(a >= b)) == sizeof(typename simd_traits<simd_t>::mask_type);
```

Comment:
This would not be the right fix for the problem. Testing here only the size of the types is wrong, as it could cause issues, if the mask type was defined not correctly but happens to have the same size.
To solve this we need to fix the mask_type. At the moment only gcc's builtin extensions are used and they only specify that the result of comparison results in a type of the same size and the same packing. But this does not mean it must be the same type. The mask_type should be decltype(a == b) in the definition of the simd_traits. I looked into the code and I think you can expose this in the simd_traits_helper and let simd_traits use this.  @marehr anything I am not seeing here?

Reply:
Mhm so I basically set mask_type to decltype(a == b)?

If `they only specify that the result of comparison results in a type of the same size and the same packing` I could also require that the sizeof the scalar_type and the length is the both for decltype(a==b) and simd_t, i.e. 
```cpp
requires (simd_traits<decltype(a == b)>::length == simd_traits<simd_t>::length) && 
               (sizeof(typename simd_traits<decltype(a == b)>::scalar_type) == 
                sizeof(typename simd_traits<simd_t>::scalar_type));
```

Reply:
No, we cannot require the mask type to be a vector. It could be a special mask type (introduced in AVX512). Any simd backend libary might implement a special type for this. Thus, simd_traits might not be valid. I think we need anyway a special concept for the mask type and then check if the mask type fulfills this concept, but I need to discuss this with @marehr.